src/AqueductV1Pair.sol

```diff
@@ -132,48 +282,100 @@ contract AqueductV1Pair is IAqueductV1Pair, AqueductV1ERC20, SuperAppBase {
         if (totalFlow0 > 0 && totalFlow1 > 0) {
             uint256 flowDirection = (totalFlow0 * reserve1) / reserve0;
             if (flowDirection > totalFlow1) {
-                int256 reserve0Diff = (int256(uint256(reserve0)) - int256(uint256(_reserve0)));
-                _reserve0 += uint112(
-                    uint256(
-                        int256(uint256(totalFlow0 * timeElapsed * LOWER_BOUND_FEE) / 10000) +
-                            (reserve0Diff -
-                                ((reserve0Diff * int112(10000 - LOWER_BOUND_FEE)) / int112(10000 - UPPER_BOUND_FEE)))
-                    )
-                );
-                _reserve1 += ((totalFlow1 * timeElapsed * LOWER_BOUND_FEE) / 10000);
+                _reserve0 +=
+                    _calculateFees(totalFlow0, timeElapsed, LOWER_BOUND_FEE) +
```

Comment:
this is now using `_calculateFees`, which doesn't do the casting: `int256(uint256(totalFlow0 * timeElapsed * LOWER_BOUND_FEE) / 10000)`. Is this safe?